src/typescript-worker/project-cache.ts

```diff
@@ -3,13 +3,12 @@ import {CachingCdn} from './caching-cdn.js';
 import { ImportMapResolver } from "./import-map-resolver";
 
 
-export const builderCache = new Map();
+export let builderInstance: TypeScriptBuilder;
```

Comment:
If we are going to implement more LSP support (completion, types, etc.), we need a LS instance for those too.

I was wondering if we should/could use the same `TypeScriptBuilder` instance for those too, since it will be up to date with the latest project files? If we do that, we might want to keep the `project-cache.ts` separate, and maybe even re-name the builder to something else.

Thoughts?

Reply:
Good point, it will probably make sense to have some separation for the reason you said. And yeah, I also agree it will probably make sense to rename it -- or maybe we'll want to factor out another class that purely encapsulate the language service parts, which is passed *in* to the builder. Something to consider in the near future!

Only thing we should do is add a license header to this file, then:

```ts
/**
 * @license
 * Copyright 2021 Google LLC
 * SPDX-License-Identifier: BSD-3-Clause
 */
```